crates/router/src/db/payment_method.rs

```diff
@@ -187,6 +215,33 @@ impl PaymentMethodInterface for MockDb {
         }
     }
 
+    async fn find_payment_method_by_customer_id_merchant_id_status(
+        &self,
+        customer_id: &str,
+        merchant_id: &str,
+        status: common_enums::PaymentMethodStatus,
+    ) -> CustomResult<Vec<storage::PaymentMethod>, errors::StorageError> {
+        let payment_methods = self.payment_methods.lock().await;
+        let payment_methods_found: Vec<storage::PaymentMethod> = payment_methods
+            .iter()
+            .filter(|pm| {
+                pm.customer_id == customer_id
+                    && pm.merchant_id == merchant_id
+                    && pm.status == status
+            })
+            .cloned()
+            .collect();
+
+        if payment_methods_found.is_empty() {
+            Err(errors::StorageError::ValueNotFound(
+                "cannot find payment methods".to_string(),
+            ))
+            .into_report()
```

Comment:
does the original implementation ( on diesel ) return this error if it finds an empty vec?

Reply:
This piece of code was copied over verbatim from the list payment methods diesel query.